File: Ruby/ShieldBooster.rb
#encoding: UTF-8

# Autores: Manuel Díaz-Meco Terrés y Ana Graciani Donaire
# 2ºDGIIM		PDOO


# Esta claserepresenta a los potenciadores de escudo que pueden tener las
# estaciones espaciales

require_relative 'ShieldToUI.rb'

module Deepspace
	class ShieldBooster

		# Constructor

		def initialize (name, boost, uses)
			# Atributos de instancia privados
			@name = name
			@boost = boost
			@uses = uses
		end

		# Método de copia (método de clase)

		def self.newCopy (s)
			new(s.name, s.boost, s.uses)
		end

		public

		# Consultores públicos
		
		# Consultor del atributo name, para así poder usar newCopy
		def name
			return @name
		end

		# Consultor del atributo boost
		def boost
			return @boost
		end

		#Consultor del atributo uses
		def uses
			return @uses
		end

		# Método de uso del escudo

		def useIt
			if (@uses > 0)
				@uses-= 1
				return @boost
			else
				return 1.0
			end
		end
		
		def to_s
			return "Name: #{@name}, Boost: #{@boost}, Uses: #{@uses}"
		end
		
		def getUIversion
			return ShieldToUI.new(self)
		end

	end
	
	# Comprobación de código
=begin
	s = ShieldBooster.new("Prueba", 2.0, 10)
	s_copy = ShieldBooster.newCopy(s)
	puts s.inspect
	puts s_copy.inspect
	puts s.name
	puts s.boost
	puts s.uses
	puts s.useIt
	puts s.uses
	puts s.to_s
=end
end

File: Java/deepspace/src/deepspace/ShieldBooster.java
/*
@authors Manuel Diaz-Meco Terrés y Ana Graciani Donaire
2ºDGIIM         PDOO


 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 * Esta clase representa a los potenciadores de escudo que pueden tener las
 * estaciones espaciales.
 */
class ShieldBooster implements CombatElement {       // Visibilidad de paquete
    
    private String name;
    private float boost;
    private int uses;
    
    // Constructor con visibilad de paquete que recibe como parámetro un
    // valor para cada uno de los atributos de la clase.
    
    ShieldBooster (String namegiven, float boostgiven, int usesgiven) {
        name = namegiven;
        boost = boostgiven;
        uses = usesgiven;
    }
    
    // Constructor copia con visibilidad de paquete.
    
    ShieldBooster (ShieldBooster s) {
        name = s.name;
        boost = s.boost;
        uses = s.uses;
    }
    
    // Consultor público para el atributo boost
    
    public float getBoost (){
        return boost;
    }
    
    // Consultor público para el atributo uses
    
    @Override
    public int getUses () {
        return uses;
    }
    
    /**
     * Si el valor del atributo uses es mayor que 0, lo decrementa en una
     * unidad y devuelve el valor del atributo boost
     * Devuelve el valor 1.0 en otro caso.
     */
    
    @Override
    public float useIt(){
        
        float returnvalue = boost;
        
        if (uses > 0)   uses--;
        else returnvalue = 1.0f;
        
        return returnvalue;
    }
    
    /**
     * Método toString
     * @return String que representa el objeto ShieldBooster
     */
    @Override
    public String toString() {
        return "name: " + name + ", boost: " + boost + ", uses: "+uses;
    }
    
    ShieldToUI getUIversion() {
        return (new ShieldToUI(this));        
    }
    
    
    
}
